lib/firefox/geckoProfiler.js

```diff
@@ -121,6 +121,14 @@ export class GeckoProfiler {
     const runner = this.runner;
     const storageManager = this.storageManager;
     const options = this.options;
+    // automatically set the index and url when using the exposed
+    // geckoprofiler commands
+    if (index == undefined && this.firefoxConfig.geckoProfilerCustom) {
+      index = 1;
+    }
+    if (url == undefined && this.firefoxConfig.geckoProfilerCustom) {
+      url = options.browsertime.url;
```

Comment:
I'm not sure this will work because sometimes we get the URL direct from the script file and we can also get it after a navigation has happened (clicking on a link etc)?

Reply:
Sorry, I mean the start/stop command looks cool, however if it will work IRL is another story :D 

I think internally the stop command needs to be able to get the correct URL and correct index, so that the geckoprofilelog will be stored at the correct place. I think you could follow the same pattern as with the screenshot command https://github.com/sitespeedio/browsertime/blob/main/lib/core/engine/command/screenshot.js - that class needs the iteration and can run JavaScript (so you can collect the current URL from the browser), Does that make sense?

Reply:
Thanks for the link, so I could use 
```javascript
const url = await this.browser
      .getDriver()
      .executeScript('return document.documentURI;');
```
to get the url.
Is there a similar way to auto grab the index?
`this.browser.options.index` I noticed this field, could I use this?

Reply:
I think you can follow the example exactly by making a new GeckoprofileCommand (or something), then when it's instantiated in the iterations, it can get the index the same ways as the screenshot class: https://github.com/sitespeedio/browsertime/blob/main/lib/core/engine/iteration.js#L176 

Then the class can have start/stop so the user more easily can handle it.

I can help tomorrow.

Reply:
Okay that makes sense now thank you!
The new push has an attempt at this implementation